pkg/hive/hive.go

```diff
@@ -127,18 +130,20 @@ func (s *Service) peersHandler(_ context.Context, peer p2p.Peer, stream p2p.Stre
 		return fmt.Errorf("close stream: %w", err)
 	}
 	for _, newPeer := range peersReq.Peers {
-		addr, err := ma.NewMultiaddr(newPeer.Underlay)
+		bzzAddress, err := bzz.ParseAddress(newPeer.Underlay, newPeer.Overlay, newPeer.Signature, s.networkID)
 		if err != nil {
-			s.logger.Infof("Skipping peer in response %s: %w", newPeer, err)
+			s.logger.Warningf("skipping peer in response %s: %w", newPeer, err)
```

Comment:
I agree, but I am not sure we should disconnect from this node unless we document that we are punishing such a behavior, as it would impose other implementations of client to pay attention and check the addresses they are sending. What I am saying, it seems to be that this should be part of the protocol then., but it is a valid point! @janos, @zelig, what do you guys think as well?

Reply:
SGTM :+1: